src/App.tsx

```diff
@@ -110,14 +120,41 @@ function HomeScreen() {
     });
   };
 
+  const submit = () => {
+    const items = Object.keys(state.items).map(itemName => {
+      const itemDetails = state.items[itemName];
+      return {
+        name: itemName,
+        cost: itemDetails.price,
+        people: itemDetails.purchaserList
+      }
+    });
+
+    try {
+      submitTransaction({
+        items, 
+        tip: state.tip,
+        tax: state.tax,
+        party: state.party
+      });
+
+      dispatch({ type: DispatchActions.RESET });
+      if (taxInputRef.current) {
+        taxInputRef.current.clear();
+      };
+    } catch {
+      // TODO: show error message
+    }
+  }
+
   return (
     <View style={styles.container}>
       <>
         <>
           <Text>Party</Text>
 
-          {state.party.map((name, index) => (
-            <Text key={`${name}-${index}`}>{name}</Text>
+          {state.party.map((name) => (
+            <Text key={`${name}`}>{name}</Text>
```

Comment:
there was a lint warning error about using the index of an array -- so i removed it, the key should still be unique if each name is unique

Reply:
doesn't have to be in this PR but in that case, do we need a check when we're adding to party list that the name we're adding is unique/needs to be unique?

or maybe we can have an id associated with each party? wonder if that lint warning is worth ignoring too if it's too much trouble lol.